### PR TITLE
TG-24: Debug/auth issues... and Refactoring

### PR DESCRIPTION
**Pull request by vararun**
Solutions to problems:
1) Headers object was using the deprecated localStorage item: 
`localStorage.getItem('cd_token')` I've updated it to use  
`JSON.parse(localStorage.getItem('sessionUser') || "{}").cd_token` 
Ideally this should not be hard coded

General Commits:
1) Minor formatting and refactoring to clean up code
2) Added linting/formatting scripts to package.json, travis.yml
3) Added frisby tests for E2E API testing
4) Add bower for front-end lib management
5) Update component css (moved base styling to [scss](http://sass-lang.com/documentation/file.SCSS_FOR_SASS_USERS.html))
6) Added Swagger Docs
7) Added PR review app generation on PR commits
8) Angular animate to replace animate.css lib dependency
9) PR review app integration with Heroku

ISSUES:
-~~API tests fail for Retrieve/Update/Delete User~~

TODO:
-Update Frisby tests, possible include in CI build?
-Update nav to close on link click

### CODE REVIEW

**Comment by javoraptor** [comment]
Can't review on cell

**Comment by vararun** [comment]
DONT NOT MERGE

File: server/models/purchase.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-'use strict';
+"use strict";
```
Comment (javoraptor) [inline server/models/purchase.js:1]: why did you undo the formatting in this commit?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+FROM docker/whalesay:latest
```
Comment (javoraptor) [inline Dockerfile:1]: have you dockerized e2e?

**Review by javoraptor — APPROVED** [review]
solid PR, go forth and merge!

File: Dockerfile (hunk 2 above)
Comment (vararun) [reply Dockerfile:1]: woops, that Dockerfile was an example one, need to remove

File: server/models/purchase.js (hunk 1 above)
Comment (vararun) [reply server/models/purchase.js:1]: I undid the changes to both the purchase and user schemas as I was getting login issues after converting it to ES6 notation. Jus need to reformat on the next branch

**Pull request merged**